library/src/main/java/com/anjlab/android/iab/v3/BillingProcessor.java

```diff
@@ -489,6 +490,42 @@ public boolean isOneTimePurchaseWithExtraParamsSupported(Bundle extraParams)
 		return isOneTimePurchaseExtraParamsSupported;
 	}
 
+	/**
+	 * Checks does API support version 6 required to request purchase history
+	 * @param type product type, accepts either {@value Constants#PRODUCT_TYPE_MANAGED}
+	 *                or {@value Constants#PRODUCT_TYPE_SUBSCRIPTION}
+	 * @return {@code true} if feature supported {@code false} otherwise
+	 */
+	public boolean isRequestBillingHistorySupported(String type) throws BillingCommunicationException
+	{
+		if (!type.equals(Constants.PRODUCT_TYPE_MANAGED) && !type.equals(Constants.PRODUCT_TYPE_SUBSCRIPTION))
+		{
+			throw new RuntimeException("Unsupported type " + type);
+		}
+
+		IInAppBillingService billing = billingService;
```

Comment:
do we really need a local var here?

Reply:
Yes, `this.billingService` becomes `null` if service disconnects.